pkg/osquery/table/user_avatar_darwin.go

```diff
@@ -0,0 +1,93 @@
+package table
+
+/*
+#cgo darwin CFLAGS: -DDARWIN -x objective-c
+#cgo darwin LDFLAGS: -framework Cocoa -framework Collaboration
+#import <Collaboration/Collaboration.h>
+#include <CoreFoundation/CoreFoundation.h>
+void Image(CFDataRef *imageDataRef, char* user) {
+NSString *userName = [NSString stringWithFormat:@"%s", user];
+CBIdentity *identity = [CBIdentity identityWithName:userName authority:[CBIdentityAuthority defaultIdentityAuthority]];
+NSImage *userImage = [identity image];
+NSData *imageData = [userImage TIFFRepresentation];
+*imageDataRef = (CFDataRef)imageData;
+}
+*/
+import "C"
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"image/png"
+	"unsafe"
+
+	osquery "github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/nfnt/resize"
+	"github.com/pkg/errors"
+	"golang.org/x/image/tiff"
+)
+
+func UserAvatar(client *osquery.ExtensionManagerClient) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("username"),
+		table.TextColumn("thumbnail"),
+	}
+	t := &userAvatarTable{client: client}
+	return table.NewPlugin("kolide_user_avatar", columns, t.generateAvatar)
+}
+
+type userAvatarTable struct {
+	client      *osquery.ExtensionManagerClient
+	primaryUser string
+}
+
+func (t *userAvatarTable) generateAvatar(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+
+	if t.primaryUser == "" {
```

Comment:
See my note above about eliminating the primary user query. Maybe we can look for all the home dirs and return all results (unless username is specified in the querycontext)?